engine/apps/alerts/tasks/notify_user.py

```diff
@@ -289,7 +290,11 @@ def perform_notification(log_record_pk):
         phone_backend.notify_by_call(user, alert_group, notification_policy)
 
     elif notification_channel == UserNotificationPolicy.NotificationChannel.TELEGRAM:
-        TelegramToUserConnector.notify_user(user, alert_group, notification_policy)
+        try:
+            TelegramToUserConnector.notify_user(user, alert_group, notification_policy)
+        except RetryAfter as e:
+            countdown = getattr(e, "retry_after", 3)
+            perform_notification.retry((log_record_pk,), countdown=countdown, exc=e)
```

Comment:
FYI, I think `retry` will raise an exception (besides queuing a new task), and since we have the `autoretry_for` set for `Exception`, you will end having 2 new tasks from this (the retry because of the exception, and the task added by the retry call). See [here](https://github.com/grafana/oncall-private/pull/2472#discussion_r1453431352) too.